### PR TITLE
Cordova: update all libraries

### PR DESCRIPTION
**Pull request by bulbigood**
Updated all Cordova libraries. Only Android is tested. Please try to build Android build by yourself (check cordova/README.md) in order to check the build and run process. You can build a signed release APK by running "build:android:signed" script.

### CODE REVIEW

File: cordova/config/config.template [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version='1.0' encoding='utf-8'?>
 <widget id="org.montelibero.solar" version="$PACKAGE_VERSION" android-versionCode="$ANDROID_VERSIONCODE" ios-CFBundleVersion="$IOS_BUNDLE_VERSION" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0" xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools">
-    <name>Solar</name>
+    <name>MTL Solar</name>
     <description>
-        Wallet for the Stellar payment network by SatoshiPay.
+        Wallet for the Stellar ecosystem by Montelibero.
     </description>
     <author email="hello@solarwallet.io" href="https://solarwallet.io/">
         SatoshiPay Ltd
```
Comment (troggy) [inline cordova/config/config.template:8]: probably need to change email, href and author name as well? Not sure where these would be used

File: cordova/package.json [hunk 2]
```diff
@@ -11,49 +11,50 @@
...
     "cordova-plugin-ionic-keyboard": "^2.2.0",
-    "cordova-plugin-local-notification": "^0.9.0-beta.2",
-    "cordova-plugin-network-information": "^2.0.1",
     "cordova-plugin-safariviewcontroller": "^2.0.0",
-    "cordova-plugin-splashscreen": "^6.0.0",
-    "phonegap-plugin-barcodescanner": "^8.0.1"
+    "cordova-plugin-splashscreen": "^6.0.0"
   },
   "devDependencies": {
+    "@moodlehq/cordova-plugin-local-notification": "^0.9.0-moodle.11",
```
Comment (troggy) [inline cordova/package.json:46]: was it moved to devDependencies on purpose?

**Comment by troggy** [comment]
@bulbigood some cordova plugins are in `devDependencies`, some in `dependencies`. Doesn't look neat

**Comment by troggy** [comment]
TLDR: did some checking, overall the app works, but I haven't tested the plugins that changed yet (see below)

✅  Android build ok (I had to adjust my environment, see below)
✅  Installed in emulator
❓ IOS build?

❓ Splashscreen worked (?), but I saw a default Cordova splash screen for a few moment before seeing Solar's (if it even was a splashscreen)
🕐 Haven't tested fingerprint plugin
🕐 Haven't tested barcodescanner plugin
🕐 Haven't tested network information plugin

Will check these on the device tomorrow.

## Few notes on a build which may be helpful for others:

1. I had to install Android Build Tools version 33 (before we had version 30) as it is a new requirement by cordova-android@12:
     - install command line tools in Android Studio
     - `~/Library/Android/sdk/cmdline-tools/latest/bin/sdkmanager "build-tools;33.0.3"`
   
2. I also had to install java 19 (as it is required by Gradle 7.6, used by cordova-android@12 ?):
     - `sudo port install openjdk19`
     - `jenv add /Library/Java/JavaVirtualMachines/openjdk19/Contents/Home`
     - `jenv local 19`
   
3. Then rebuild the android platform like `rm -rf cordova/platforms/android/ && cd cordova && npm run install:android`
   
4. When installing android platform you may see this error now:

   ```
   Plugin doesn't support this project's cordova-android version. cordova-android: 12.0.1, failed version requirement: >=3.6.0 <11.0.0
   Skipping 'cordova-plugin-splashscreen' for android
   ```

   The reason: `cordova-plugin-splashscreen` plugin is [not needed](https://github.com/apache/cordova-plugin-splashscreen/issues/344#issuecomment-1580596870) for Android anymore as it is now [part of cordova-android](https://cordova.apache.org/docs/en/12.x/core/features/splashscreen/index.html#android-specific-information) anymore. But still needed for iOS. I guess we can just ignore the  error.

File: cordova/package.json (hunk 2 above)
Comment (bulbigood) [reply cordova/package.json:46]: No, this file is operated by Cordova CLI. What I did was "cordova plugin remove <plugin>" and "cordova plugin add <plugin>" with a couple of plugins. It's better to trust Cordova, I think.

**Comment by bulbigood** [comment]
> 2. I also had to install java 19 (as it is required by Gradle 7.6, used by cordova-android@12 ?):

Based on the compatibility matrix of Gradle, it should be noted that the specified Java versions are the maximum supported ones. I recommend using Java 11 or Java 17, as they are LTS versions. Also, I found many complaints on the topic of upgrading to the latest Cordova version, and the recommendations to update to Java 11.

**Comment by troggy** [comment]
> I recommend using Java 11

@bulbigood I was using java 11 (openjdk11 specifically) prior to this PR and wasn't able to build the app with it.

**Review by troggy — CHANGES_REQUESTED** [review]
Second take of android testing:
✅ Android build ok (I had to adjust my environment, see [here](https://github.com/Montelibero/mtl_solar/pull/25#issuecomment-1933187694))
✅ Installed in emulator
✅ Fingerprint plugin works fine (checked unlock)
✅ Network information plugin (checked offline/online detection)
✅ Barcode scanner works fine (had problems on the first run though.. probably related to my phone? cannot reproduce since then)
🛑 Splash screen now shows default Cordova splash screen for a few moments. Checked the original Solar — it wasn't the case
🛑 Notification plugin seems not be working. On the previous codebase Solar is asking for permission to send notifications on the first run. On this PR I didn't get this request
🛑  Transparent status bar is not working anymore — it is (https://github.com/Montelibero/mtl_solar/assets/163447/cbc7007c-adc0-4f21-990d-959d88d9d423 now. [How it was before](https://github.com/Montelibero/mtl_solar/assets/163447/fa983b45-5e62-4957-9f2a-3e38a89691de)

Not tested:
- cordova-plugin-device (minor change)
- cordova-plugin-proguard

File: cordova/config/config.template (hunk 1 above)
Comment (bulbigood) [reply cordova/config/config.template:8]: I'll roll back the changes here
